lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/launch/JavaLaunchCommandBuilder.java

```diff
@@ -37,19 +37,22 @@
  * Build the launch command for Java REEF processes.
  */
 public final class JavaLaunchCommandBuilder implements LaunchCommandBuilder {
+
   private static final Logger LOG = Logger.getLogger(JavaLaunchCommandBuilder.class.getName());
 
-  private static final String DEFAULT_JAVA_PATH = System.getenv("JAVA_HOME") + "/bin/" + "java";
+  private static final String DEFAULT_JAVA_PATH = "java"; // Assume java binary is in the PATH
```

Comment:
This is tricky. What if `java` is not on the path or if the right `java` is not on the path? Can we check how Hadoop  and Spark handle this and follow their lead? That way, we confuse users the least.